apps/native/app/src/screens/document-detail/document-detail.tsx

```diff
@@ -197,29 +205,35 @@ export const DocumentDetailScreen: NavigationFunctionComponent<{
   })
 
   const Document = {
-    ...(docRes.data?.getDocument || {}),
-    ...doc.data,
+    ...(doc?.data || {}),
+    ...(docRes.data?.documentV2 || {}),
+    ...(docContent.data?.documentV2 || {}),
   }
 
   useEffect(() => {
     if (doc.missing) {
-      void getListDocuments({
-        variables: {
-          input: {
-            page: 1,
-            pageSize: 50,
-          },
-        },
-      })
+      // If we don't have the document in the cache, fetch it
+      void getDocument()
+    } else if (doc.data?.content === null) {
+      // Content is missing, fetch it
+      void getDocumentContent()
     }
-  }, [doc])
+  }, [doc.missing, doc.data?.content, docId])
 
   const [visible, setVisible] = useState(false)
   const [loaded, setLoaded] = useState(false)
   const [pdfUrl, setPdfUrl] = useState('')
   const [touched, setTouched] = useState(false)
-  const hasPdf = Document.fileType?.toLocaleLowerCase() === 'pdf'
-  const isHtml = typeof Document.html === 'string' && Document.html !== ''
+
+  const loading = docRes.loading || !accessToken
+  const fileTypeLoaded =
+    !docContent.loading && !docContent.error && !docRes.loading && !docRes.error
```

Comment:
This is probably `true` during initial render. Simpler to check if `Document?.content?.type` has a value?